nntrainer/tensor/cl_operations/cl_kernels/int4_gemv.cl

```diff
@@ -102,6 +116,54 @@
 #define DT_FILTER_BLOCK_READ8(ptr, offset) BLOCK_READN(char, 8, ptr, offset)
 #define DT_FILTER_BLOCK_READ16(ptr, offset) BLOCK_READN(char, 16, ptr, offset)
 
+// ==========================================================================
+// Block-read emulation (when Intel block-read intrinsics aren't present).
+// ==========================================================================
+#define BLOCK_READ_IMPL_1 ret = ptr[idx];
+
+#define BLOCK_READ_IMPL_2                                                      \
+  ret.s0 = ptr[idx];                                                           \
+  idx += get_max_sub_group_size();                                             \
+  ret.s1 = ptr[idx];                                                           \
+  idx += get_max_sub_group_size();
```

Comment:
Be careful with such #define.

Someone may do...:
```
if (a == b)
  BLOCK_READ_IMPL_2;
else
  BLOCK_READ_IMPL_4;
```
which will break everything.

https://stackoverflow.com/questions/923822/whats-the-use-of-do-while0-when-we-define-a-macro